src/ConnectedMode.UnitTests/Migration/ConnectedModeMigrationTests.cs

```diff
@@ -23,49 +23,70 @@
 using System.Threading;
 using System.Threading.Tasks;
 using SonarLint.VisualStudio.ConnectedMode.Migration;
-using SonarLint.VisualStudio.Integration;
-using SonarLint.VisualStudio.TestInfrastructure;
 using SonarLint.VisualStudio.ConnectedMode.UnitTests.Migration.ConnectedModeMigrationTestsExtensions;
-using SonarLint.VisualStudio.ConnectedMode.Binding;
 using SonarLint.VisualStudio.Core.Binding;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.TestInfrastructure;
 
 namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.Migration
 {
     [TestClass]
     public class ConnectedModeMigrationTests
     {
+        private static readonly LegacySettings DefaultTestLegacySettings = new LegacySettings("folder", "cs ruleset", "cs xml", "vb ruleset", "vb xml");
+
         private static BoundSonarQubeProject AnyBoundProject = new BoundSonarQubeProject(new Uri("http://localhost:9000"), "any-key", "any-name");
 
         [TestMethod]
         public void MefCtor_CheckTypeIsNonShared()
             => MefTestHelpers.CheckIsNonSharedMefComponent<ConnectedModeMigration>();
 
         [TestMethod]
-        public void MefCtor_CheckIsExported_NoProjectCleaners()
+        public void MefCtor_CheckIsExported()
         {
             MefTestHelpers.CheckTypeCanBeImported<ConnectedModeMigration, IConnectedModeMigration>(
+                MefTestHelpers.CreateExport<IMigrationSettingsProvider>(),
                 MefTestHelpers.CreateExport<IFileProvider>(),
                 MefTestHelpers.CreateExport<IFileCleaner>(),
                 MefTestHelpers.CreateExport<IVsAwareFileSystem>(),
                 MefTestHelpers.CreateExport<ILogger>());
         }
 
+        [TestMethod]
+        public async Task Migrate_ExpectedOldBindingIsPassedSettingsProvider()
+        {
+            var fileProvider = CreateFileProvider();
+
+            var settings = new LegacySettings("expected folder", "any", "any", "any", "any");
+            var oldBinding = new BoundSonarQubeProject(new Uri("http://any"), "expected project key", "any");
+            var settingsProvider = CreateSettingsProvider(settings);
+
+            var testSubject = CreateTestSubject(fileProvider.Object, settingsProvider: settingsProvider.Object);
+
+            await testSubject.MigrateAsync(oldBinding, null, CancellationToken.None);
+
+            settingsProvider.Verify(x => x.GetAsync("expected project key"), Times.Once);
+        }
+
         [TestMethod]
         public async Task Migrate_NoFilesToClean_DirectoryIsDeleted()
         {
             var fileProvider = CreateFileProvider();
             var fileSystem = new Mock<IVsAwareFileSystem>();
             var fileCleaner = new Mock<IFileCleaner>();
 
-            var testSubject = CreateTestSubject(fileProvider.Object, fileCleaner.Object, fileSystem.Object);
+            var settings = new LegacySettings("expected folder", "any", "any", "any", "any");
```

Comment:
Fixed up the existing tests since the class being tested now needs a settings provider.